tests/api/id/[form]/bearer.test.js

```diff
@@ -15,144 +17,312 @@ jest.mock("../../../../lib/integration/dbConnector", () => {
+        expect.objectContaining({ token: "toekakdnaodk" })
+      );
     });
 
-    await retrieve(req, res);
-    expect(res.statusCode).toBe(200);
-    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ token: "toekakdnaodk" }));
-  });
+    it("Should return a 403 status code if there is no valid session", async () => {
+      client.getSession.mockReturnValue(undefined);
+
+      const { req, res } = createMocks({
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        query: {
+          form: "45",
+        },
+      });
 
-  it("Shouldn't allow this request without a valid session", async () => {
-    client.getSession.mockReturnValue(undefined);
-
-    const { req, res } = createMocks({
-      method: "GET",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      query: {
-        form: "45",
-      },
+      await retrieve(req, res);
+      expect(res.statusCode).toBe(403);
+      expect(JSON.parse(res._getData())).toEqual(
+        expect.objectContaining({ error: "Access Denied" })
+      );
     });
 
-    await retrieve(req, res);
-    expect(res.statusCode).toBe(403);
-    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ error: "Access Denied" }));
-  });
+    it("Should return 404 status code if no form was found", async () => {
+      const mockSession = {
+        expires: "1",
+        user: { email: "admin@cds.ca", name: "Admin user", image: "null" },
+      };
+      client.getSession.mockReturnValue(mockSession);
+      // Mocking executeQuery to return an empty array
+      executeQuery.mockReturnValue({ rows: [], rowCount: 0 });
+
+      const { req, res } = createMocks({
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        query: {
+          form: "23",
+        },
+      });
+      await retrieve(req, res);
 
-  it("Should return 404 statusCode Not Found if an empty [] value was returned", async () => {
-    const mockSession = {
-      expires: "1",
-      user: { email: "admin@cds.ca", name: "Admin user", image: "null" },
-    };
-    client.getSession.mockReturnValue(mockSession);
-    // Mocking executeQuery to return an empty array
-    executeQuery.mockReturnValue({ rows: [], rowCount: 0 });
-
-    const { req, res } = createMocks({
-      method: "GET",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      query: {
-        form: "23",
-      },
+      expect(res.statusCode).toBe(404);
+      expect(JSON.parse(res._getData()).error).toEqual("Not Found");
     });
-    await retrieve(req, res);
 
-    expect(res.statusCode).toBe(404);
-    expect(JSON.parse(res._getData()).error).toEqual("Not Found");
+    it("Should return 500 status code if there's an unexpected error", async () => {
+      const mockSession = {
+        expires: "1",
+        user: { email: "admin@cds.ca", name: "Admin user", image: "null" },
```

Comment:
`@cds.ca` is an actual domain that has nothing to do with CDS-SNC. Even though nothing is being sent to this email address, perhaps we need an actual test email?

Reply:
Its just the mocked session. There are no side effects that are actually using this email